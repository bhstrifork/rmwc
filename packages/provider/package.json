{
  "name": "@rmwc/provider",
  "version": "0.0.1",
  "description": "RMWC Provider component",
  "main": "./index.js",
  "types": "./index.d.ts",
  "exports": {
    ".": {
      "import": "./index.mjs",
      "require": "./index.js"
    }
  },
  "peerDependencies": {
    "react": ">=16.8.x",
    "react-dom": ">=16.8.x"
  },
  "dependencies": {
    "@rmwc/base": "^14.0.1-alpha.0",
    "@rmwc/types": "^14.0.1-alpha.0"
  }
}
