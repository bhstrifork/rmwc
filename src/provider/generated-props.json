{
  "npm": {},
  "typescript": {
    "useProviderContext": {
      "fileName": "src/provider/index.tsx",
      "flags": {
        "isExternal": false,
        "isOptional": false,
        "isPrivate": false,
        "isProtected": false,
        "isPublic": false,
        "isRest": false,
        "isStatic": false
      },
      "kind": "method",
      "name": "useProviderContext",
      "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L82",
      "signatures": [
        {
          "fileName": "src/provider/index.tsx",
          "kind": "signature",
          "name": "useProviderContext",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L82",
          "parameters": [],
          "returnType": "RMWCProviderProps",
          "type": "() => RMWCProviderProps"
        }
      ]
    },
    "RMWCProvider": {
      "documentation": {
        "contents": [
          "<p>A provider for setting global options in RMWC.</p>\n"
        ],
        "contentsRaw": "A provider for setting global options in RMWC.",
        "metadata": {}
      },
      "fileName": "src/provider/index.tsx",
      "flags": {
        "isExternal": false,
        "isOptional": false,
        "isPrivate": false,
        "isProtected": false,
        "isPublic": false,
        "isRest": false,
        "isStatic": false
      },
      "kind": "method",
      "name": "RMWCProvider",
      "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L85",
      "signatures": [
        {
          "documentation": {
            "contents": [
              "<p>A provider for setting global options in RMWC.</p>\n"
            ],
            "contentsRaw": "A provider for setting global options in RMWC.",
            "metadata": {}
          },
          "fileName": "src/provider/index.tsx",
          "kind": "signature",
          "name": "RMWCProvider",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L85",
          "parameters": [
            {
              "flags": {
                "isExternal": false,
                "isOptional": false,
                "isPrivate": false,
                "isProtected": false,
                "isPublic": false,
                "isRest": false,
                "isStatic": false
              },
              "kind": "parameter",
              "name": "__namedParameters",
              "type": ""
            }
          ],
          "returnType": "Element<>",
          "type": "(__namedParameters: ) => Element<>"
        }
      ]
    },
    "Readme": {
      "fileName": "src/provider/readme.tsx",
      "flags": {
        "isExternal": false,
        "isOptional": false,
        "isPrivate": false,
        "isProtected": false,
        "isPublic": false,
        "isRest": false,
        "isStatic": false
      },
      "kind": "method",
      "name": "Readme",
      "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/readme.tsx#L9",
      "signatures": [
        {
          "fileName": "src/provider/readme.tsx",
          "kind": "signature",
          "name": "Readme",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/readme.tsx#L9",
          "parameters": [],
          "returnType": "Element<>",
          "type": "() => Element<>"
        }
      ]
    },
    "RMWCProviderProps": {
      "documentation": {
        "contents": [
          "<p>A provider for setting global options in RMWC.</p>\n"
        ],
        "contentsRaw": "A provider for setting global options in RMWC.",
        "metadata": {}
      },
      "fileName": "src/provider/index.tsx",
      "flags": {
        "isExternal": false,
        "isOptional": false,
        "isPrivate": false,
        "isProtected": false,
        "isPublic": false,
        "isRest": false,
        "isStatic": false
      },
      "kind": "interface",
      "name": "RMWCProviderProps",
      "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L47",
      "methods": [],
      "properties": [
        {
          "documentation": {
            "contents": [
              "<p>Children to render</p>\n"
            ],
            "contentsRaw": "Children to render",
            "metadata": {}
          },
          "fileName": "src/provider/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": true,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "children",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L57",
          "type": "React.ReactNode"
        },
        {
          "documentation": {
            "contents": [
              "<p>Global options for icons</p>\n"
            ],
            "contentsRaw": "Global options for icons",
            "metadata": {}
          },
          "fileName": "src/provider/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": true,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "icon",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L51",
          "type": "Partial<RMWC.IconOptions>"
        },
        {
          "documentation": {
            "contents": [
              "<p>Enable / Disable interaction ripples globally</p>\n"
            ],
            "contentsRaw": "Enable / Disable interaction ripples globally",
            "metadata": {}
          },
          "fileName": "src/provider/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": true,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "ripple",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L49",
          "type": "undefined | false | true"
        },
        {
          "documentation": {
            "contents": [
              "<p>Global tooltip options</p>\n"
            ],
            "contentsRaw": "Global tooltip options",
            "metadata": {}
          },
          "fileName": "src/provider/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": true,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "tooltip",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L53",
          "type": "Partial<TooltipOptions>"
        },
        {
          "documentation": {
            "contents": [
              "<p>Global typography options</p>\n"
            ],
            "contentsRaw": "Global typography options",
            "metadata": {}
          },
          "fileName": "src/provider/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": true,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "typography",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/provider/index.tsx#L55",
          "type": "Partial<TypographyOptions>"
        }
      ]
    }
  }
}
