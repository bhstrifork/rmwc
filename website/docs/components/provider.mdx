---
sidebar_label: Provider
title: ''
---

import Provider from '@rmwc/provider/README.mdx';

<Provider />
