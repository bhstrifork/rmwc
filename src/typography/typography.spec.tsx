import React from 'react';
import { render, screen } from '@testing-library/react';
import { Typography } from './';
import { RMWCProvider } from '../provider';

describe('Typography', () => {
  it('renders', () => {
    const { asFragment } = render(<Typography use="body1" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with each type', () => {
    const { asFragment } = render(
      <>
        <Typography use="headline1" />
        <Typography use="headline2" />
        <Typography use="headline3" />
        <Typography use="headline4" />
        <Typography use="headline5" />
        <Typography use="headline6" />
        <Typography use="subtitle1" />
        <Typography use="subtitle2" />
        <Typography use="body1" />
        <Typography use="body2" />
        <Typography use="caption" />
        <Typography use="button" />
        <Typography use="button" />
      </>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('can have custom classnames', () => {
    const { container } = render(
      <Typography use="body1" className={'my-custom-classname'} />
    );
    expect(container.firstChild).toHaveClass('my-custom-classname');
  });

  it('supports refs', () => {
    const myRef = React.createRef<HTMLElement>();

    render(
      <Typography use="body1" ref={myRef}>
        Hello
      </Typography>
    );

    expect(myRef.current).toBeTruthy();
  });

  it('works with RMWCProvider', () => {
    render(
      <RMWCProvider
        typography={{
          /** Make all Typography components default to <div>  */
          defaultTag: 'div',
          /** Make headline5 <h5>  */
          headline5: 'h5',
          /** Make body2 <p>  */
          body2: 'p',
          /** Use your own component  */
          body1: ({ children, ...rest }) => (
            <span>
              <b>{children}!!!</b>
            </span>
          )
        }}
      >
        <Typography use="headline6">Rendered default `div`</Typography>
        <Typography use="headline5">Rendered with `h5`</Typography>
        <Typography use="body2">Rendered with `p`</Typography>
        <Typography use="body1">Custom rendering</Typography>
      </RMWCProvider>
    );

    render(
      <RMWCProvider typography={undefined}>
        <Typography use="headline6">Rendered default `div` 2</Typography>
      </RMWCProvider>
    );
    expect(screen.getByText('Rendered default `div`')).toBeInTheDocument();
    expect(screen.getByText('Rendered with `p`')).toBeInTheDocument();
    expect(screen.getByText('Rendered default `div` 2')).toBeInTheDocument();
  });
});
