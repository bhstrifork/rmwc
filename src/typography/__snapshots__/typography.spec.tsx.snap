// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography renders 1`] = `
<DocumentFragment>
  <span
    class="mdc-typography--body1"
  />
</DocumentFragment>
`;

exports[`Typography renders with each type 1`] = `
<DocumentFragment>
  <span
    class="mdc-typography--headline1"
  />
  <span
    class="mdc-typography--headline2"
  />
  <span
    class="mdc-typography--headline3"
  />
  <span
    class="mdc-typography--headline4"
  />
  <span
    class="mdc-typography--headline5"
  />
  <span
    class="mdc-typography--headline6"
  />
  <span
    class="mdc-typography--subtitle1"
  />
  <span
    class="mdc-typography--subtitle2"
  />
  <span
    class="mdc-typography--body1"
  />
  <span
    class="mdc-typography--body2"
  />
  <span
    class="mdc-typography--caption"
  />
  <span
    class="mdc-typography--button"
  />
  <span
    class="mdc-typography--button"
  />
</DocumentFragment>
`;
